src/averageAges.js

```diff
@@ -37,7 +38,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = (withChildren)
+    ? people.filter(female => female.sex === 'f'
+      && people.some(child => child.mother === female.name))
+    : people.filter(female => female.sex === 'f');
```

Comment:
One of possible ways 
```suggestion
  const women = people.filter(female => withChildren 
    ? female.sex === 'f' && people.some(child => child.mother === female.name)
    : female.sex === 'f'
  );
```